### PR TITLE
[Merged by Bors] - doc(Tactic): fix more mathlib3 names in doc comments

### PR DESCRIPTION
**Pull request by grunweg**
Auto-generated using a script, with manual fix-ups.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Attr/Register.lean [hunk 1]
```diff
@@ -81,5 +81,5 @@ discharge theorems about the trivial case (where we know `Subsingleton α` and m
 in e.g. groups are trivially true). -/
 register_simp_attr nontriviality
 
-/-- A stub attribute for `is_poly`. -/
+/-- A stub attribute for `IsPoly`. -/
```
Comment (grunweg) [inline Mathlib/Tactic/Attr/Register.lean:84]: This is a mis-match; shouldn't this attribute be renamed instead?
I don't see this attribute actually used in practice; according to comments, it seems unused. I'm fine reverting this, but wanted to call attention to this one.

File: Mathlib/Tactic/Nontriviality/Core.lean [hunk 2]
```diff
@@ -55,7 +55,7 @@ def nontrivialityByAssumption (g : MVarId) : MetaM Unit := do
 
 /-- Attempts to generate a `Nontrivial α` hypothesis.
 
-The tactic first looks for an instance using `infer_instance`.
+The tactic first looks for an instance using `inferInstance`.
```
Comment (grunweg) [inline Mathlib/Tactic/Nontriviality/Core.lean:58]: I'm not sure if this is a correct change or not. Happy to revert.

File: Mathlib/Tactic/ToAdditive.lean [hunk 3]
```diff
@@ -90,7 +90,7 @@ that have previously been labeled with `to_additive`.
 
 In the `mul_comm'` example above, `to_additive` maps:
 * `mul_comm'` to `add_comm'`,
-* `comm_semigroup` to `add_comm_semigroup`,
+* `comm_semigroup` to `AddCommSemigroup`,
```
Comment (grunweg) [inline Mathlib/Tactic/ToAdditive.lean:93]: I guess all the docs in this file ought to be updated; happy to split that into a separate PR.

**Comment by grunweg** [comment]
Let me remove the "easy" tag: I don't think this is particularly hard, but it's not *totally* obvious either.

File: Mathlib/Tactic/ToAdditive.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Tactic/ToAdditive.lean:93]: @fpvandoorn

File: Mathlib/Tactic/Attr/Register.lean (hunk 1 above)
Comment (kmill) [reply Mathlib/Tactic/Attr/Register.lean:84]: It should be named however is needed to get the syntax for the attribute to be in snake case. Looking at the source for `register_label_attr`, it seems that `is_poly` is correct here.

File: Mathlib/Tactic/Nontriviality/Core.lean (hunk 2 above)
Comment (kmill) [reply Mathlib/Tactic/Nontriviality/Core.lean:58]: It could go either way, since `infer_instance` is the tactic and `inferInstance` is the term.

It's probably better to avoid the question entirely and improve the docstring:
```suggestion
The tactic first checks to see that there is not already a `Nontrivial α` instance
before trying to synthesize one using other techniques.
```

File: Mathlib/Tactic/Attr/Register.lean (hunk 1 above)
Comment (grunweg) [inline Mathlib/Tactic/Attr/Register.lean:84]: Thank you for the clarification.
```suggestion
/-- A stub attribute for `is_poly`. -/

**Comment by grunweg** [comment]
Filed #11951 for the ToAdditive changes.

File: Mathlib/Tactic/ToAdditive.lean [hunk 4]
```diff
@@ -1355,7 +1355,7 @@ Examples:
 * `@Mul.mul Nat n m` (i.e. `(n * m : Nat)`) will not change to `+`, since its
   first argument is `Nat`, an identifier not applied to any arguments.
 * `@Mul.mul (α × β) x y` will change to `+`. It's first argument contains only the identifier
-  `prod`, but this is applied to arguments, `α` and `β`.
+  `Prod`, but this is applied to arguments, `α` and `β`.
```
Comment (grunweg) [inline Mathlib/Tactic/ToAdditive.lean:1358]: Is this a desirable change? (I'm unsure.) If undesirable:
```suggestion
  `prod`, but this is applied to arguments, `α` and `β`.

File: Mathlib/Tactic/ToAdditive.lean [hunk 5]
```diff
@@ -106,7 +106,7 @@ Examples:
...
```
Comment (grunweg) [inline Mathlib/Tactic/ToAdditive.lean:109]: I'm not sure if this change is desirable. If not, here's a revert:
```suggestion
  `prod`, but this is applied to arguments, `α` and `β`.

File: Mathlib/Tactic/Congr!.lean [hunk 6]
```diff
@@ -46,7 +46,7 @@ structure Congr!.Config where
   By default this is `.reducible`, which prevents unfolding of most definitions. -/
   preTransparency : TransparencyMode := TransparencyMode.reducible
   /-- For passes that synthesize a congruence lemma using one side of the equality,
-  we run the pass both for the left-hand side and the right-hand side. If `preferLHS` is `true`
+  we run the pass both for the left-hand side and the right-hand side. If `preferLHS` is `True`
```
Comment (fpvandoorn) [inline Mathlib/Tactic/Congr!.lean:49]: `Bool.true` is exported to the root namespace, so `true` is the Lean 4 name for the value of `Bool`, so this is a bad fix (also elsewhere). `true`/`false` should only be capitalized if the type is meant.

File: Mathlib/Tactic/ToAdditive.lean (hunk 5 above)
Comment (fpvandoorn) [reply Mathlib/Tactic/ToAdditive.lean:109]: This one is good

File: Mathlib/Tactic/Congr!.lean (hunk 6 above)
Comment (grunweg) [reply Mathlib/Tactic/Congr!.lean]: Now this makes sense, thank you! I have hopefully fixed all instances.

File: Mathlib/Tactic/FailIfNoProgress.lean [hunk 7]
```diff
@@ -37,7 +37,7 @@ syntax (name := failIfNoProgress) "fail_if_no_progress " tacticSeq : tactic
 
 /-- `lctxIsDefEq l₁ l₂` compares two lists of `Option LocalDecl`s (as returned from e.g.
 `(← (← getMainGoal).getDecl).lctx.decls.toList`). It returns `true` if they have the same
-local declarations in the same order (up to defeq, without setting mvars), and `false` otherwise.
+local declarations in the same order (up to defeq, without setting mvars), and `False` otherwise.
```
Comment (fpvandoorn) [inline Mathlib/Tactic/FailIfNoProgress.lean:40]:
```suggestion
local declarations in the same order (up to defeq, without setting mvars), and `false` otherwise.
```

**Review by fpvandoorn — COMMENTED** [review]
bors d+

**Comment by fpvandoorn** [comment]
bors merge
